pkg/build/tls.go

```diff
@@ -0,0 +1,200 @@
+package build
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/big"
+	"time"
+
+	"github.com/cnoe-io/idpbuilder/globals"
+	"github.com/cnoe-io/idpbuilder/pkg/k8s"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	certificateOrgName = "cnoe.io"
+)
+
+func createIngressCertificateConfigMap(ctx context.Context, kubeClient client.Client, cert []byte) error {
+	cm := &corev1.ConfigMap{
```

Comment:
Is there a reason to use a CM and not a secret ?

Reply:
We do use secret for ingress-nginx. Since this function only deals with certificate, not privatekey, I don't think it's necessary to use a secret here. Certificates are usually distributed in CMs. 

Having said that, it may actually make sense to use a secret if we are to use ESO, at least until they add support for CMs.

Reply:
updated to create secret instead.